core/src/main/kotlin/endpoints/plugin/DrillPluginWs.kt

```diff
@@ -29,51 +29,65 @@ class DrillPluginWs(override val kodein: Kodein) : KodeinAware, Sender {
     private val cacheService: CacheService by instance()
     private val eventStorage: Cache<String, String> by cacheService
     private val sessionStorage: ConcurrentMap<String, MutableSet<SessionData>> = ConcurrentHashMap()
-    private val agentManager: AgentManager by instance()
 
-    @Suppress("UNCHECKED_CAST")
     override suspend fun send(agentId: String, buildVersion: String, destination: Any, message: Any) {
-        val dest = destination as? String ?: app.toLocation(destination)
-        val id = "$agentId:$dest:$buildVersion"
-
-        if (message.toString().isEmpty()) {
+        val dest = destination.toDestination(app)
+        val id = CacheId(
+            agentId = agentId,
+            destination = dest,
+            buildVersion = buildVersion
+        )
+
+        if (isRemoveMessage(message)) {
```

Comment:
leave it without method extraction, add //TODO will be removed after api changes